airbyte-cdk/python/airbyte_cdk/sources/declarative/parsers/factory.py

```diff
@@ -243,7 +243,7 @@ def is_object_definition_with_class_name(definition):
 
     @staticmethod
     def is_object_definition_with_type(definition):
-        return isinstance(definition, dict) and "type" in definition
+        return isinstance(definition, dict) and "type" in definition and definition["type"] != "object"
```

Comment:
the name of the method `is_object...` should this check be `!=`?

Reply:
The problem with how are connector specifications are defined, the `type` field is an overloaded term. In the manifest, `type` is just shorthand so we don't have to include the full classpath. But for the connector specification object, `type: object` is actually part of the spec schema. So the factory gets confused and thinks we need to parse the spec as another declarative component, but we really rant this to remain a Mapping. Without this line, it looks for `object` in the CLASS_TYPES_REGISTRY

It's a little bit of a hack, but it allows us to make the manifest defined and external defined specs interchangable

Reply:
@brianjlai  should we rename our `type` field to something less overloaded as part of lowcode to beta?

Reply:
can you add this context as a comment?

Reply:
yup will add!